### PR TITLE
Add `cwd` to the manifest as a command annotation

### PR DESCRIPTION
**Pull request by maminrayej**
Adds support for specifying the working directory for each command as a wasi annotation. For example:

```toml
[command.annotations.wasi]
cwd = "/data"
```

### CODE REVIEW

File: lib/wasix/src/runners/wasi.rs [hunk 1]
```diff
@@ -353,6 +353,10 @@ impl crate::runners::Runner for WasiRunner {
             }
         }
 
+        if let Some(cwd) = cmd.metadata().annotation::<String>("cwd")? {
```
Comment (theduke) [inline lib/wasix/src/runners/wasi.rs:356]: Shouldn't this also be a `wasi` annotation?
Not every kind of runner would definitely have a CWD, and it fits together with the other wasi settings.

Comment (theduke) [inline lib/wasix/src/runners/wasi.rs:356]: Also, shouldn't we add this to the type definitions in webc?

Comment (maminrayej) [reply lib/wasix/src/runners/wasi.rs:356]: It was added as a custom annotation which does not have a type definition and is just a plain `Value`, but adding it as a wasi annotation will require a change to webc.

File: tests/integration/cli/tests/packages/list-cwd/wasmer.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+[fs]
+"/data" = "."
+
+[[command]]
+name = "run"
+module = "main"
+runner = "wasi"
+
+[command.annotations]
+cwd = "/data"
```
Comment (syrusakbary) [inline tests/integration/cli/tests/packages/list-cwd/wasmer.toml:14]: This needs to be part of the WASI annotations, not just the command

**Review by syrusakbary — APPROVED** [review]

**Pull request merged**